metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchMetaAlertDao.java

```diff
@@ -614,8 +625,15 @@ protected void calculateMetaScores(Document metaAlert) {
       }
       metaScores = new MetaScores(scores);
     }
+
+    // add a summary (max, min, avg, count, sum) of all the threat scores from the child alerts
     metaAlert.getDocument().putAll(metaScores.getMetaScores());
-    metaAlert.getDocument().put(threatTriageField, metaScores.getMetaScores().get(threatSort));
+
+    // the overall threat score for the metaalert; either max, min, avg, count or sum of all child scores
```

Comment:
why isn't it a float to start with?  isn't that the real issue?

Reply:
The calculations were done as Double and given to ES.  However, there's no definition of the field in ES (It just used automatic mapping), so it was given the ES double.

Reply:
The `ElasticsearchMetaAlertDao` adds an overall threat score to the Metaalert.  The overall threat score can be any one of the following summary aggregations of the child alerts; sum, min, max, count, average, or median.  These summary values are calculated in `MetaScores` and result in Double values.  Since the other sensor indices currently define `threat:triage:score` as a float, this solution just casts this to a Double to match those.

I think an alternative way to solve this is to just make the `threat:triage:score` in each of the sensor indices a Double as you mentioned.  I think your approach seems a little cleaner to me.  Although, I am not sure if there are other down sides I am not thinking about.  Can anyone else think of a problem with this approach? @justinleet @merrimanr ?

Reply:
I would not hold up the PR for this point if it is too much.  Just seems that we are coding around something else.

Reply:
I think it is really just a matter of what we'd expect a user to define as scores for their threat triage rules.  Are they really going to define values greater than a 32-bit float?  Or values that sum to greater than a 32-bit float?

I think it is a good point to discuss, Otto.  I'd suggest we go with the float approach now, as it minimizes the scope of change in this PR.  But we can revisit whether a double should be used after we migrate to ES 5.x.